Fix: Test data contamination in report service due to hardcoded store_code in cleanup fixture

## Problem

The test `test_terminal_id_filtering_with_multi_terminal_data` in `services/report/tests/test_terminal_id_parsing.py` was failing with:
```
AssertionError: Terminal 5555 should have 1 transaction(s), got 3
```

### Root Cause

The `clean_test_data` fixture in `services/report/tests/conftest.py` was hardcoded to clean up data with `store_code="STORE001"`, but many tests (including terminal_id_parsing tests) use `store_code="5678"` from the environment variable. This caused:

1. Test data from `test_setup_data.py` (store_code="5678") was not being cleaned up
2. Subsequent tests saw leftover data, causing assertion failures
3. Tests that use `store_code="STORE001"` (like test_edge_cases.py) would fail if the fixture was changed to only use the environment variable

## Solution

Modified the `clean_test_data` fixture to clean BOTH store codes:
- Environment variable store_code ("5678") - used by test_terminal_id_parsing.py and test_setup_data.py
- Hardcoded "STORE001" - used by test_edge_cases.py and other tests

### Changes Made

**File: services/report/tests/conftest.py (lines 124-150)**
```python
# Clean both environment variable store_code and hardcoded test store codes
# Some tests use env var ("5678"), others use hardcoded values ("STORE001")
store_code_env = os.environ.get("STORE_CODE")
test_store_codes = [store_code_env, "STORE001"]

# Clean up for both store codes
total_deleted = 0
for store_code in test_store_codes:
    delete_result = await collection.delete_many({
        "tenant_id": tenant_id,
        "store_code": store_code
    })
    total_deleted += delete_result.deleted_count
```

**File: services/report/tests/test_terminal_id_parsing.py**
- Line 177: Added `clean_test_data` fixture parameter to ensure clean database state
- Lines 298-301: Added 3-second sleep to allow pub/sub event processing

## Test Results

All 20 report service tests now pass:
- ✓ test_terminal_id_parsing.py (including the previously failing test)
- ✓ test_edge_cases.py 
- ✓ All other report service tests

## Impact

This fix ensures proper test isolation and prevents test data contamination across the entire report service test suite.

--- a/services/report/tests/conftest.py
+++ b/services/report/tests/conftest.py
@@ -121,22 +121,33 @@ async def clean_test_data(set_env_vars):
     tran_repo = TranlogRepository(db, tenant_id)
     collection = db[tran_repo.collection_name]
 
+    # Clean both environment variable store_code and hardcoded test store codes
+    # Some tests use env var ("5678"), others use hardcoded values ("STORE001")
+    store_code_env = os.environ.get("STORE_CODE")
+    test_store_codes = [store_code_env, "STORE001"]
+
     # IMPORTANT: MongoDB stores fields in snake_case (tenant_id, store_code)
     # because we use model_dump() without by_alias=True
-    delete_result = await collection.delete_many({
-        "tenant_id": tenant_id,
-        "store_code": "STORE001"
-    })
-    print(f"[CLEANUP BEFORE] Deleted {delete_result.deleted_count} test documents before test")
+    total_deleted = 0
+    for store_code in test_store_codes:
+        delete_result = await collection.delete_many({
+            "tenant_id": tenant_id,
+            "store_code": store_code
+        })
+        total_deleted += delete_result.deleted_count
+    print(f"[CLEANUP BEFORE] Deleted {total_deleted} test documents before test (store_codes: {test_store_codes})")
 
     yield
 
     # Teardown: clean data after test to prevent contamination of next test
-    delete_result_after = await collection.delete_many({
-        "tenant_id": tenant_id,
-        "store_code": "STORE001"
-    })
-    print(f"[CLEANUP AFTER] Deleted {delete_result_after.deleted_count} test documents after test")
+    total_deleted_after = 0
+    for store_code in test_store_codes:
+        delete_result_after = await collection.delete_many({
+            "tenant_id": tenant_id,
+            "store_code": store_code
+        })
+        total_deleted_after += delete_result_after.deleted_count
+    print(f"[CLEANUP AFTER] Deleted {total_deleted_after} test documents after test")
 
 
 @pytest_asyncio.fixture(scope="function", autouse=True)

--- a/services/report/tests/test_terminal_id_parsing.py
+++ b/services/report/tests/test_terminal_id_parsing.py
@@ -174,7 +174,7 @@ async def test_terminal_id_format_validation():
 
 
 @pytest.mark.asyncio()
-async def test_terminal_id_filtering_with_multi_terminal_data(http_client):
+async def test_terminal_id_filtering_with_multi_terminal_data(http_client, clean_test_data):
     """
     Test that terminal_id filtering returns only data for the specified terminal.
 
@@ -294,6 +294,11 @@ async def test_terminal_id_filtering_with_multi_terminal_data(http_client):
     )
     assert response.status_code == status.HTTP_201_CREATED
     print(f"✓ Created transaction {tran_no_2} for terminal {terminal_no_2}")
+
+    # Wait for pub/sub events to be processed
+    import asyncio
+    print("\nWaiting for transactions to be processed...")
+    await asyncio.sleep(3)
 
     # Request TERMINAL-specific report with terminal_id parameter
     # This tests that terminal_id is correctly parsed and used for filtering

